libs/geometry/src/transform.rs

```diff
@@ -456,7 +456,29 @@ fn unitary_matinv(a: &[[i8; 2]; 2]) -> [[i8; 2]; 2] {
     [[a[1][1], -a[0][1]], [-a[1][0], a[0][0]]]
 }
 
-/// A trait for specifying how an object is changed by a transformation.
+/// A trait for specifying how an object is changed by a [`Transformation`].
+pub trait TransformRef: TranslateRef {
+    /// Applies matrix-vector [`Transformation`] `trans`.
+    fn transform_ref(&self, trans: Transformation) -> Self;
+}
+
+impl TransformRef for () {
+    fn transform_ref(&self, _trans: Transformation) -> Self {}
+}
+
+impl<T: TransformRef> TransformRef for Vec<T> {
+    fn transform_ref(&self, trans: Transformation) -> Self {
+        self.iter().map(|elt| elt.transform_ref(trans)).collect()
+    }
+}
+
+impl<T: TransformRef> TransformRef for Option<T> {
+    fn transform_ref(&self, trans: Transformation) -> Self {
+        self.as_ref().map(move |elt| elt.transform_ref(trans))
+    }
+}
+
+/// A trait for specifying how an object is changed by a [`Transformation`].
 #[impl_for_tuples(32)]
 pub trait TransformMut {
```

Comment:
Why does TransformMut not need TranslateMut then?

Reply:
added TranslateMut as a supertrait of TransformMut (and similarly for Translate/Transform).